### PR TITLE
Initial commits

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request contains the initial commits for this Ansible role.

## 💭 Motivation and context ##

We need to be able to install Terraform for cisagov/cool-system#204.

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Stellar! 🤩

File: tasks/main.yml [hunk 1]
```diff
@@ -1,2 +1,34 @@
 ---
-# tasks file for skeleton
+# This is necessary in order to unarchive the Terraform zip and the
+# terraform-docs tarball.
+- name: Install unzip
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
- name: Install tar and unzip
```

File: tasks/main.yml [hunk 2]
```diff
@@ -1,2 +1,34 @@
...
+  ansible.builtin.package:
+    name:
+      - tar
+      - unzip
+
+- name: Download and install terraform binary
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
- name: Download and install Terraform binary
```

File: defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# be installed
+install_dir: /usr/local/bin
+# The group of the Terraform and terraform-docs binaries
+group: root
+# The mode of the Terraform and terraform-docs binaries
+mode: 0766
+# The owner of the Terraform and terraform-docs binaries
+owner: root
+# The version of Terraform to install
+terraform_version: 1.0.3
```
Comment (mcdonnnj) [inline defaults/main.yml]: Just a thought since this version was released 2021-08-04.

```suggestion
terraform_version: 1.0.4
```

**Review by mcdonnnj — APPROVED** [review]
Straightforward implementation 👍 Just as a thought; would it make sense to install using the package manager directions Hashicorp has? They show support for `apt`/`rpm` for Ubuntu/Debian, CentOS/RHEL, Fedora, and Amazon Linux. This would allow an easier in-place upgrade path.

**Comment by jsf9k** [comment]
> Straightforward implementation  Just as a thought; would it make sense to install using the package manager directions Hashicorp has? They show support for `apt`/`rpm` for Ubuntu/Debian, CentOS/RHEL, Fedora, and Amazon Linux. This would allow an easier in-place upgrade path.

There are packages for the distributions you listed, but Hashicorp reserves the right to drop support for future versions on any of the older versions of those distributions as it sees fit.  It's also the case that as new versions of those distributions come out, packages available for older versions of Terraform may not be available on those new versions.  Given the "already fully linked" feature of `go` executables, installing via the archives seemed like an easier path to trod.

**Comment by mcdonnnj** [comment]
> 
> 
> > Straightforward implementation  Just as a thought; would it make sense to install using the package manager directions Hashicorp has? They show support for `apt`/`rpm` for Ubuntu/Debian, CentOS/RHEL, Fedora, and Amazon Linux. This would allow an easier in-place upgrade path.
> 
> There are packages for the distributions you listed, but Hashicorp reserves the right to drop support for future versions on any of the older versions of those distributions as it sees fit. It's also the case that as new versions of those distributions come out, packages available for older versions of Terraform may not be available on those new versions. Given the "already fully linked" feature of `go` executables, installing via the archives seemed like an easier path to trod.

That's reasonable. Just thought it was worth a mention so it was clear on the "why" for installing from downloaded zip.

**Pull request merged**